audbackend/core/utils.py

```diff
@@ -78,6 +80,66 @@ def check_version(version: str) -> str:
     return version
 
 
+def checksum(file: str) -> str:
+    r"""Checksum of file.
+
+    The checksum is given by the MD5 sum
+    as calculated with :func:`audeer.md5`.
+
+    As parquet files are stored non-deterministically,
+    we allow to use them with precalculated checksums,
+    stored under the key ``b"hash"`` in its metadata.
+    To support this feature pyarrow_
+    has to be installed.
+    A deterministic checksum,
+    based on the content of the parquet file,
+    can be calculated with :func:`audformat.utils.hash`.
+    If the key is not present in the metadata of the parquet file,
+    or pyarrow_ is not installed,
+    its MD5 sum is calculated instead.
+
+    .. _pyarrow: https://arrow.apache.org/docs/python/index.html
+
+    Args:
+        file: file path with extension
+
+    Returns:
+        MD5 checksum of file
+
+    Raises:
+        FileNotFoundError: if ``file`` does not exist
+
+    Examples:
+        >>> checksum("src.txt")
+        'd41d8cd98f00b204e9800998ecf8427e'
+        >>> import audformat
+        >>> import pandas as pd
+        >>> import pyarrow as pa
+        >>> import pyarrow.parquet as pq
+        >>> df = pd.DataFrame([0, 1], columns=["a"])
+        >>> hash = audformat.utils.hash(df, strict=True)
+        >>> hash
+        '9021a9b6e1e696ba9de4fe29346319b2'
+        >>> table = pa.Table.from_pandas(df)
+        >>> table = table.replace_schema_metadata({"hash": hash})
+        >>> pq.write_table(table, "file.parquet", compression="snappy")
+        >>> checksum("file.parquet")
+        '9021a9b6e1e696ba9de4fe29346319b2'
+
+    """
+    ext = audeer.file_extension(file)
+    if ext == "parquet":
+        try:
+            import pyarrow.parquet as parquet
+
+            metadata = parquet.read_schema(file).metadata or {}
+            if b"hash" in metadata:
```

Comment:
I'm not sure if this can happen. I tried to create test cases where `hash` was `None` or `123`, but I was not able to create such an parquet file as
```python
table.replace_schema_metadata({"hash": None})
```
fails with
```
TypeError: expected bytes, NoneType found
```
and
```python
table.replace_schema_metadata({"hash": 123})
```
fails with
```
TypeError: expected bytes, int found
```

So maybe it is ok to assume, that this cannot happen?

Reply:
True. There is no binary of string representation of `None`, so this cannot happen. 
Great that there are new test cases covering this. 

Resolving this.